os/net/socket/net_sockets.c

```diff
@@ -133,10 +136,36 @@ void net_initlist(FAR struct socketlist *list)
 
 void net_releaselist(FAR struct socketlist *list)
 {
-	DEBUGASSERT(list);
+	int i;
+	int ret;
+	struct lwip_sock *psock;
+
+	for (i = 0; i < CONFIG_NSOCKET_DESCRIPTORS; ++i) {
+		if (list->sl_sockets[i].sock) {
+
+			psock = (struct lwip_sock *)list->sl_sockets[i].sock;
+
+			if (list->sl_sockets[i].s_crefs == 1) {
+				ret = lwip_sock_close(psock);
+				if (ret) {
+					ndbg("socket could not close properly\n");
+					list->sl_sockets[i].sock = NULL;
+					(void)sem_destroy(&list->sl_sem);
```

Comment:
You could replace this with a "break" statement to avoid duplication?

Reply:
can you please clear it more?
where is duplication?